### DISCUSSION TITLE
How is calc compared with GNU bc?

### DISCUSSION THREAD

**Discussion by ghost** [lcn2/calc#94] [Q&A] [answered]
Github suggested your project to me. I'm currently using GNU bc.

**Comment by lcn2** [comment, marked as answer]
Calc is a C-like language with a rich command line as well as interactive interface.

Calc competes files internally as rational numbers a/b, or and complex rational values.

So:

```calc
; sqrt(2)
	1.4142135623730950488
```

will give you a few digits of the value of the square root of 2.

If you wish to see more digits, you can extend the number of digits displayed we well an the precision (called epsilon):

```calc
; display(100)
; sqrt(2,1e-100)
        1.4142135623730950488016887242096980785696718753769480731766797379907324784621070388503875343276415727
```

Most functions take an epsilon argument. See:

```calc
; help sqrt
```

You can set the global epsilon, so if you want 1000 digits of precision:

```calc
; display(1000)
; epsilon(1e-1000)
; sqrt(2)
; sin(1/3)
; etc.
```

Calc understands complex numbers, so the log of 3 * sqrt(-1):

```calc
; log(3i)
```

You can set values in your own calc start fille (usually ~/.calcrc).  See:

```calc
; help environment
```

Calc as lists, arrays, associative arrays, lists, list, user defined data types, etc.

Calc as a custom facility for people to compile in their own custom functions.  A number of supercomputers with special hardware use calc via the custom interface.

You can compile your own C code that can link to and use the calc library.  See `sample_many.c` and `sample_rand.c` for examples.

Like bc, calc is "free as in beer".  Calc is shipped under LGPL v2 so you can use, give away, add, ship, sell or give away calc as you wish as long as others can do the same.

Calc has has been around since 1984.   We accept code contributions:

```calc
; help contrib
```

**Share and enjoy!  :-)**

**Comment by lcn2** [comment]
Calc as a fish set of FILE I/O operations, unlike bc.   See:

```calc
; man file
```

One may set the output base to a number of different baes:

```calc
; help base
```

You can configure and customize calc as well:

```calc
; man config
```

Calc comes with lots of bulletin functions:

```calc
; man builtin
```

To find out about a given bulletin functions, such as append or random or ssq:

```calc
; man append
; man random
; man ssq
```

While calc may have inspires of the Unix v7 `bc(`)` command, and there are differences, it is bc and more.
